docs/source/markdown/podman.1.md

```diff
@@ -99,6 +99,9 @@ and "$HOME/.config/cni/net.d" as rootless. CNI is deprecated from Podman in the
 
 Redirect stdout to /dev/null. This command prevents all stdout from the Podman command. The **--noout**  option is not block stderr or stdout from containers.
 
+#### **--out**=*path*
+Redirect stdout to the specified path. This command allows redirecting all stdout from the Podman command which can be used to capture its output directly to a file. It can also be used to suppress output by specifying /dev/null as the path. The **--out** option will not affect stderr or stdout from containers.
```

Comment:
This is hard to understand. First, `This command` is wrong (I realize that you copied it from `--noout`; that is wrong, and should never have been approved like that, and it's something our team will have to fix). And the `will not affect...from containers`, ouch, you're right, and as a user I find that **super** confusing: it means `podman --out /tmp/foo run ...` violates POLA. (So does `--nout`). I think that should be made very very clear in the documentation; I regret that I don't have any idea how to do so.

Reply:
Yes, this was definitely copy-pasted from noout's documentation. I threw it together really quick so that the DCO check would pass as I originally wrote this logic against 4.5.0 over a month ago and wanted to see the results of my actual tests. I definitely could use more input on how to better word this before I squash all the history out of my commits.

In my opinion, though, noout not only had this weird behaviour but it was also broken. The original implementation was opening up /dev/null as non-writable, so not broken in that it didn't do what was intended...but broken in that if you strace you can see EBADFD returned by those components that are trying to write to that fd.

The first test in 001-basic.bats (line 285), tests for this brokenness...but imo, it's a terrible test as it captures both stdout + stderr instead of checking just stderr directly. This is possible with the `--separate-stderr` bats option (1.5), but it looked like it required me to update `run_podman` (system/helpers.bash) to add that ability. Perhaps there's a better way to test for podman being unable to write to stdout, though.

To be honest, I only discovered "--noout" while looking for an option to let me write an id directly to a file w/o depending on the existence of a shell. There are many places where podman writes well-formed data to stdout that could be considered separate from container output (such as just an id, with `--format`, or where it just echo's the commandline parameter). So the idea was to promote the already-existing `--noout` option to `--out` in order to enable podman to write this formatted output to a file. This way the already-existing functionality of `--noout` is the same as `--out /dev/null`. noout has been there since #11515, though, and updated at #16201. So perhaps others are already depending on noout to silence podman's output.

Reply:
Would it help to modify the documentation so that it doesn't mention stderr or containers at all.. instead suggesting its intended usage for atomically capturing output from podman's commands that emit identifiers and container information? This way the independence from container output is better distinguished maybe?

Reply:
you can see that the current --noout is broken by running
```
$ podman --noout unshare ls
ls: write error: Bad file descriptor
```
---
I think we should fix the `podman run` bug as shown above, i.e. move this: https://github.com/containers/podman/blob/86746c067fe20435bc05bb5656ead6ff48adfcfd/cmd/podman/containers/run.go#L52-L54
down in the run() function:

Then just document that the container output is still logged and can be viewed with podman logs later, to disable the container logs you need to run with `--log-driver none`.

Reply:
Awesome. Thanks for answering my questions, @Luap99. I simplified the relevant test to use your suggestion. One thing to note is that the `podman-unshare` test needed to be constrained by "skip_if_remote" (to avoid podman-remote), but I'm sure that's fine.

I also moved the assignment of `OutputStream`, `InputStream`, and `ErrorStream` to right before this code which sets up attachments to the individual streams.

https://github.com/containers/podman/blob/86746c067fe20435bc05bb5656ead6ff48adfcfd/cmd/podman/containers/run.go#L164-L168

To clarify your suggestion about the documentation, "--log-driver none" seems to only be available in `podman-run`. As this is a global option, the "--log-driver" option isn't mentioned in the relevant manpage for `podman(1)`. Still, I tried to incorporate both of your concerns so that it reads:

> Redirect the output of podman to the specified path without affecting the container output or its logs. This parameter can be used to capture the output from any of podman's commands directly into a file and enable suppression of podman's output by specifying /dev/null as the path. To explicitly disable the container logging, the **--log-driver** option should be used.

Does that seem fine?